src/gui/log/loglistview.h

```diff
@@ -26,35 +27,27 @@
  * exception statement from your version.
  */
 
-#ifndef LOGLISTWIDGET_H
-#define LOGLISTWIDGET_H
+#pragma once
 
-#include <QListWidget>
-#include "base/logger.h"
+#include <QListView>
 
-class QKeyEvent;
+#include "base/logger.h"
 
-class LogListWidget : public QListWidget
+class LogListView : public QListView
 {
     Q_OBJECT
+    Q_DISABLE_COPY(LogListView)
 
 public:
-    // -1 is the portable way to have all the bits set
-    explicit LogListWidget(int maxLines, const Log::MsgTypes &types = Log::ALL, QWidget *parent = nullptr);
-    void showMsgTypes(const Log::MsgTypes &types);
+    LogListView(QWidget *parent = nullptr);
 
-public slots:
-    void appendLine(const QString &line, const Log::MsgType &type);
+signals:
+    void resetModel();
 
-protected slots:
-    void copySelection();
+private slots:
+    void displayListMenu(const QPoint &pos);
+    void copySelection() const;
 
 protected:
```

Comment:
Seems this can go private.

Reply:
done

Reply:
done? you didn't change anything here...

Reply:
they should be private slots, displayListmenu is connected to context menu requested and copy selection is connected to copy action

Reply:
I mean shouldn't `keyPressEvent()` be private?

Reply:
sorry, done